jib-core/src/main/java/com/google/cloud/tools/jib/JibSystemProperties.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2018 Google LLC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import java.util.function.Function;
+
+/** Names of system properties defined/used by Jib. */
+public class JibSystemProperties {
+
+  @VisibleForTesting public static final String HTTP_TIMEOUT = "jib.httpTimeout";
+
+  @VisibleForTesting
+  public static final String SEND_CREDENTIALS_OVER_HTTP = "sendCredentialsOverHttp";
+
+  private static final String SERIALIZE = "jibSerialize";
+
+  private static final String DISABLE_USER_AGENT = "_JIB_DISABLE_USER_AGENT";
+
+  /**
+   * Gets the HTTP connection/read timeouts for registry interactions in milliseconds.
+   *
+   * @return the value of the {@code jib.httpTimeout} system property
+   */
+  public static Integer getHttpTimeout() {
```

Comment:
Null pointer exception in a bunch of places if I do this.

Reply:
Oh, then this should be `@Nullable` or return an `Optional` to reflect nullability, and probably indicate that this can return null if no timeout is set. Another option is to choose a sensible default timeout for indicating no timeout set.

Reply:
I think we mention in the readme that the default is 20000, so I think I'll just return that if it's not set.